src/validations/html/rules/MismatchedTag.js

```diff
@@ -0,0 +1,139 @@
+import {List, Map, Record, Set} from 'immutable';
+import Code from '../rules/Code';
+
+const Location = Record({row: 0, column: 0});  // eslint-disable-line new-cap
+
+export default class {
```

Comment:
Hey @tenyoung795 so are we talking about something like the below?

```html
<body>
This is a paragraph.</p>

<p>Here is another one
</body>
```

In that case I think it’s probably better behavior to just treat this as two errors: first, an unmatched closing tag, then later, an unclosed opening tag. Try the above markup in this branch and see if you agree that the current error is a bit mystifying.

Obviously if a student generates this situation by cut-and-pasting they may have indeed intended to have the closing tag close the opening tag, but in that case I think it’s still better to show the two errors to get the students thinking more directly about what the correct HTML structure is.